controller/volume_controller.go

```diff
@@ -2916,7 +2916,22 @@ func (vc *VolumeController) checkForAutoDetachment(v *longhorn.Volume, e *longho
 			break
 		}
 	}
-	if !(e.Spec.RequestedBackupRestore != "" && e.Spec.RequestedBackupRestore == e.Status.LastRestoredBackup &&
+
+	// make sure the backup volume has been synced at least once
+	if backupVolumeName, isExist := v.Labels[types.LonghornLabelBackupVolume]; isExist && backupVolumeName != "" {
+		backupVolume, err := vc.ds.GetBackupVolumeRO(backupVolumeName)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get backup volume: %v", v.Name)
+		}
+		if !(backupVolume.Status.LastSyncedAt.After(backupVolume.Spec.SyncRequestedAt.Time) ||
+			backupVolume.Status.LastSyncedAt.Equal(backupVolume.Spec.SyncRequestedAt.DeepCopy())) {
```

Comment:
NIT: How about `backupVolume.Status.LastSyncedAt.Before(backupVolume.Spec.SyncRequestedAt.Time)`?

Reply:
Thanks. I think it will make it look intuitive.